### PR TITLE
man: stop recommending putting myhostname after dns

### PR DESCRIPTION
**Pull request by flokli**
nss-resolve also looks in /etc/hosts, and has the same local hostname
resolving logic as nss-myhostname. We shouldn't recommend another order
than nss-resolve uses internally.

When nss-resolve is used, there's no possibility to override
nss-myhostname hosts via DNS *anyway*.

On top of that, it's not a good idea to allow DNS to override local
hostnames as all - at least not something we should advertise in the
docs.

Followup of f918c67d38ba6ccd4eb0dc657f3f3155e5010cae /
https://github.com/systemd/systemd/pull/16754.

### CODE REVIEW

**Comment by keszybz** [comment]
Yeah, I think it's reasonable, but there was strong opposition in the past.

**Comment by flokli** [comment]
> Yeah, I think it's reasonable, but there was strong opposition in the past.

I assume if you want to override `bar.localhost`, you can probably still do - but then likely without having `nss-resolve` in that list either.

**Comment by poettering** [comment]
please keep in mind that nss-myhostname does two things: resolves names → ip address, and ip addresses → names. However, for both directions we can only define one order of modules. And that sucks, since fixing things in the wrong order means that IP addresses would always be resolved via our module instead of via DNS even though various projects assume that resolving thelocal IP address would yield the DNS name of the host.

Not sure if this is a biggie, but this deserves to be discussed, and probably be explicitly documented.

We could also compile the module twice if needed, and depending on which precise name it is called under only implement one or the other direction of resolution.

**Comment by keszybz** [comment]
> Not sure if this is a biggie, but this deserves to be discussed, and probably be explicitly documented.

I think we should just document this. With `nss-resolve` inserted earlier, this is just the fallback path anyway. And I don't think we should make the fallback path different than the main path. Also, the whole scheme is too complicated already… In real use people want to insert other modules like nss-mdns and others, and we have a hard time doing this correctly.

**Comment by flokli** [comment]
I pushed an additional commit that adds this remark to both nss-resolve and nss-myhostname.

**Comment by keszybz** [comment]
This doesn't build. Please use `build/man/man nss-myhostname` and `build/man/html nss-myhostname` to verify how the page looks rendered.

Also, please remove the empty lines. They will be ignored when parsing xml, and misleadingly look like separate paragraphs. We generally reflow the text approximately as will be rendered in the man page.

File: man/nss-resolve.xml [hunk 1]
```diff
@@ -52,6 +52,11 @@
     it is still recommended (see examples below) to keep <command>nss-myhostname</command> configured in
     <filename>/etc/nsswitch.conf</filename>, to keep those names resolveable if
     <command>systemd-resolved</command> is not running.</para>
+
+    Please keep in mind that <command>nss-myhostname<command>
+    (and <command>nss-resolve</command>) also resolve in the other direction -
```
Comment (keszybz) [inline man/nss-resolve.xml]: `-` → `—`

**Comment by flokli** [comment]
I addressed the two remarks. The manpages now build, the misleading newlines are gone, and the text looks similar to how it rendered in my terminal (modulo the lines with a lot of markup)

**Comment by keszybz** [comment]
LGTM, thanks.

**Comment by flokli** [comment]
@keszybz Thanks! Can this also be cherry-picked into systemd-stable?

I assume this all already applies to the current stable release, too.

**Comment by keszybz** [comment]
Sure, I'll include it in the next batch.

**Pull request merged**